### PR TITLE
expression: implement vectorized evaluation for `builtinCastStringAsDurationSig`

### PR DESCRIPTION
**Pull request by SilvaXiang**
### What problem does this PR solve? 

expression: implement vectorized evaluation for `builtinCastStringAsDurationSig` from #12176 
### What is changed and how it works?

2.7% faster than before:

    BenchmarkVectorizedBuiltinCastFunc/builtinCastStringAsDurationSig-VecBuiltinFunc-4                   660           1755315 ns/op          114721 B/op       9203 allocs/op
    BenchmarkVectorizedBuiltinCastFunc/builtinCastStringAsDurationSig-NonVecBuiltinFunc-4                650           1805265 ns/op          114722 B/op       9203 allocs/op
    

### Check List 

Tests 

 - Unit test

### CODE REVIEW

**Comment by SilvaXiang** [comment]
> Sorry for my late reply, there are some problems, please fix them:
> [2019-12-04T06:56:28.906Z] FAIL: integration_test.go:1157: testIntegrationSuite2.TestTimeBuiltin
> [2019-12-04T06:56:28.906Z] 
> [2019-12-04T06:56:28.906Z] integration_test.go:1553:
> [2019-12-04T06:56:28.906Z]     //for hour
> [2019-12-04T06:56:28.906Z]     result = tk.MustQuery(`SELECT hour("12:13:14.123456"), hour("12:13:14.000010"), hour("272:59:55"), hour(020005), hour(null), hour("27aaaa2:59:55");`)
> [2019-12-04T06:56:28.906Z] /home/jenkins/agent/workspace/tidb_ghpr_check_2/go/src/github.com/pingcap/tidb/util/testkit/testkit.go:270:
> [2019-12-04T06:56:28.906Z]     tk.c.Check(err, check.IsNil, comment)
> [2019-12-04T06:56:28.906Z] ... value *errors.withStack = input does not match format ("input does not match format")
> [2019-12-04T06:56:28.906Z] ... sql:SELECT hour("12:13:14.123456"), hour("12:13:14.000010"), hour("272:59:55"), hour(020005), hour(null), hour("27aaaa2:59:55");, args:[]
> [2019-12-04T06:56:28.906Z] 
> [2019-12-04T06:56:28.906Z] integration_test.go:1554:
> [2019-12-04T06:56:28.906Z]     result.Check(testkit.Rows("12 12 272 2 <nil> <nil>"))
> [2019-12-04T06:56:28.906Z] /home/jenkins/agent/workspace/tidb_ghpr_check_2/go/src/github.com/pingcap/tidb/util/testkit/testkit.go:62:
> [2019-12-04T06:56:28.906Z]     res.c.Assert(resBuff.String(), check.Equals, needBuff.String(), res.comment)
> [2019-12-04T06:56:28.906Z] ... obtained string = ""
> [2019-12-04T06:56:28.906Z] ... expected string = "[12 12 272 2 <nil> <nil>]\n"
> [2019-12-04T06:56:28.906Z] ... sql:SELECT hour("12:13:14.123456"), hour("12:13:14.000010"), hour("272:59:55"), hour(020005), hour(null), hour("27aaaa2:59:55");, args:[]

Could you give me some instructions how to fix them?
These problem confuse me for a long time......

**Comment by francis0407** [comment]
I'm working on this.

**Comment by francis0407** [comment]
The unit test fails due to the incorrect implementation of vectorized `builtinHour`. https://github.com/pingcap/tidb/pull/13949 is fixing this.

**Comment by SilvaXiang** [comment]
> The unit test fails due to the incorrect implementation of vectorized `builtinHour`. #13949 is fixing this.

但是那个解决了后,我这边还过不了.....

**Comment by francis0407** [comment]
builtinMinuteSig, builtinSecondSig, builtinMicroSecondSig also have the same problem as #13949 fixed. I will open a PR to fix them.

**Comment by francis0407** [comment]
@SilvaXiang fail to build, please fix it. I think we can pass the unit test this time.

**Comment by francis0407** [comment]
Hi @SilvaXiang, please run `go fmt` in expression package.

File: expression/builtin_cast_vec.go [hunk 1]
```diff
@@ -14,15 +14,13 @@
...
-	"strings"
-
-	"github.com/pingcap/errors"
 	"github.com/pingcap/parser/mysql"
 	"github.com/pingcap/tidb/types"
 	"github.com/pingcap/tidb/types/json"
 	"github.com/pingcap/tidb/util/chunk"
+	"math"
+	"strconv"
+	"strings"
```
Comment (francis0407) [inline expression/builtin_cast_vec.go]: Move this to the top of the import list:
```golang
"math"
"strconv"
"strings"

"github.com/...."
....
```

Comment (SilvaXiang) [reply expression/builtin_cast_vec.go]: 运行go fmt后就是这样，如果move上去后过不了....

Comment (francis0407) [reply expression/builtin_cast_vec.go]: Hmmmm, it might be a bug on Windows. I will open a PR to your repo to help you fix this.

Comment (francis0407) [reply expression/builtin_cast_vec.go]: You can try to merge this PR https://github.com/SilvaXiang/tidb/pull/1

File: expression/builtin_cast_vec_test.go [hunk 2]
```diff
@@ -51,7 +51,6 @@ var vecBuiltinCastCases = map[string][]vecExprBenchCase{
 		{retEvalType: types.ETDuration, childrenTypes: []types.EvalType{types.ETInt}, geners: []dataGenerator{new(randDurInt)}},
 		{retEvalType: types.ETDuration, childrenTypes: []types.EvalType{types.ETReal}, geners: []dataGenerator{new(randDurReal)}},
 		{retEvalType: types.ETDuration, childrenTypes: []types.EvalType{types.ETDecimal}, geners: []dataGenerator{new(randDurDecimal)}},
-		{retEvalType: types.ETDuration, childrenTypes: []types.EvalType{types.ETString}, geners: []dataGenerator{new(randDurString)}},
```
Comment (francis0407) [inline expression/builtin_cast_vec_test.go]: Why remove this? We need this test case actually.

**Pull request merged**